test/blackbox-tests/test-cases/pkg/opam-var/opam-var-global.t

```diff
@@ -0,0 +1,107 @@
+  $ . ./opam-var-helpers.sh
+
+Here we test the translation and implementation of global opam variables.
+This also includes switch specific variables since these are all trivial for Dune.
+opam-version
+  $ test opam-version
+  (build
+   (run echo %{pkg-self:opam-version}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-35:
+  4 |  (run echo %{pkg-self:opam-version}))
+                 ^^^^^^^^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+jobs
+  $ test_lockfile jobs 
+  (build
+   (run echo %{jobs}))
+make
+  $ where_is_make="$(dirname $(which make))" 
+  > export BUILD_PATH_PREFIX_MAP="MAKE_DIR=$where_is_make:$BUILD_PATH_PREFIX_MAP"
+  $ test make 
+  (build
+   (run echo %{make}))
+  MAKE_DIR/make
+arch
+  $ export ARCH=$(uname -m)
+  $ test arch 2>&1 | sed "s/^$ARCH/ARCH/g"  
+  (build
+   (run echo %{arch}))
+  ARCH
+os
+  $ test os
+  (build
+   (run echo %{pkg-self:os}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-25:
+  4 |  (run echo %{pkg-self:os}))
+                 ^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+os-distribution
+  $ test os-distribution
+  (build
+   (run echo %{os_distribution}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-29:
+  4 |  (run echo %{os_distribution}))
+                 ^^^^^^^^^^^^^^^^^^
+  Error: Unknown variable %{os_distribution}
+  [1]
+os-family
+  $ test os-fmaily
+  (build
+   (run echo %{pkg-self:os-fmaily}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-32:
+  4 |  (run echo %{pkg-self:os-fmaily}))
```

Comment:
`os-family`

Reply:
Do you want me to fix how these are translated? The correct opam variable is being tested, its just our translation that is wrong.